### PR TITLE
Adds ip6-detect script

### PR DESCRIPTION
**Pull request by GoelDeepak**
## High-level description

Adds ip6-detect scripts for aws and azure. These scripts are similar to what we have for ipv4. 
In phase-1, these scripts would only be used by mesos-agent to be able to populate NetworkInfo for the containers that would be running on IPv6 in host mode.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-1044](https://jira.mesosphere.com/browse/DCOS_OSS-1044) Add an ip6-detect script to advertise IPv6 address on the host.

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-1735](https://jira.mesosphere.com/browse/DCOS_OSS-1735) IPv6 for Docker Containers Only

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [x] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/cloudformation/simple  https://jira.mesosphere.com/browse/DCOS-18825

**Comment by orsenthil** [comment]
I overrode the status of `teamcity/dcos/test/aws/cloudformation/simple` as the cloudformation tests were not even starting in multiple attempts. The EE counterpart had the clouformation based install tested.

**Comment by GoelDeepak** [comment]
@asridharan we cannot do that because that would require all the cluster to be launched on IPv6 AMI whereas all our integration test clusters are running IPv4

**Comment by asridharan** [comment]
@GoelDeepak agreed. The ask here is not to test this in our current integration tests (though will have to figure out a way to do it) but to have a flag that when ip6_detect script is set its setup in the agent flags. Otherwise the agents can't consume this information. 

In other words unlike the `ip_detect` script the `ip6_detect` script should be optional.

**Comment by GoelDeepak** [comment]
@asridharan ok got it. Let me see how that could be achieved

**Comment by orsenthil** [comment]
@mesosphere-mergebot label work in progress.

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

**Comment by GoelDeepak** [comment]
@asridharan could you take a look at it?

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -8,7 +8,12 @@ package:
     content: |
       {
         "environment": {
+{% switch enable_mesos_ipv6_discovery %}
+{% case "false" %}
```
Comment (asridharan) [inline gen/dcos-config.yaml]: The `MESOS_IP_DISCOVERY_COMMAND` should be independent of the `MESOS_IPV6_DISCOVERY_COMMAND`, i.e. even if the `MESOS_IPV6_DISCOVERY_COMMAND` is set the `MESOS_IP_DISCOVERY_COMMAND` should still be required. Mesos agents only listen to IPv4 addresses at this point. They take in an IPv6 address, but they never actually open an IPv6 socket. So we always need to specify and IPv4 address for the agent.

File: gen/ip-detect/aws6.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
```
Comment (asridharan) [inline gen/ip-detect/aws6.sh]: Did we test this out on AWS?

File: gen/ip-detect/azure6.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
```
Comment (asridharan) [inline gen/ip-detect/azure6.sh]: Did we test this out on Azure?

File: gen/ip-detect/vagrant6.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (asridharan) [inline gen/ip-detect/vagrant6.sh]: Can we test this out with `dcos-vagrant`?

File: gen/ip-detect/aws6.sh (hunk 2 above)
Comment (GoelDeepak) [reply gen/ip-detect/aws6.sh]: yes

File: gen/ip-detect/azure6.sh (hunk 3 above)
Comment (GoelDeepak) [reply gen/ip-detect/azure6.sh]: yes

File: gen/ip-detect/vagrant6.sh (hunk 4 above)
Comment (GoelDeepak) [reply gen/ip-detect/vagrant6.sh]: yes

File: gen/dcos-config.yaml (hunk 1 above)
Comment (GoelDeepak) [reply gen/dcos-config.yaml]: I don't think that's true. If I specify MESOS_IPV6_DISCOVERY_COMMAND without a flag then cluster doesn't come up if its IPV4 only

Comment (asridharan) [reply gen/dcos-config.yaml]: Did you check the cluster for the error you are seeing when you are specifying both these flags? I think you are specifying this flag for the master as well? The `MESOS_IPV6_DISCOVERY_COMMAND` flag exists only for the agent. So my guess is that if you specify it both for the master and the agent, the master might not be coming up.

File: gen/ip-detect/aws6.sh (hunk 2 above)
Comment (asridharan) [reply gen/ip-detect/aws6.sh]: Thanks !!

File: gen/ip-detect/vagrant6.sh (hunk 4 above)
Comment (asridharan) [reply gen/ip-detect/vagrant6.sh]: Thanks !!

File: gen/ip-detect/azure6.sh (hunk 3 above)
Comment (asridharan) [reply gen/ip-detect/azure6.sh]: Thanks !!

File: gen/dcos-config.yaml (hunk 1 above)
Comment (asridharan) [reply gen/dcos-config.yaml]: Just to confirm this the output from my local build:
```
[vagrant@centos8 bin]$ ./mesos-master.sh --help | grep discovery
                                                                                 with `--ip_discovery_command`.
  --ip_discovery_command=VALUE                                                   Optional IP discovery binary: if set, it is expected to emit
[vagrant@centos8 bin]$ ./mesos-agent.sh --help | grep discovery
  --appc_simple_discovery_uri_prefix=VALUE                         URI prefix to be used for simple discovery of appc images,
                                                                   with `--ip_discovery_command`.
                                                                   with '--ip6_discovery_command'.
  --ip6_discovery_command=VALUE                                    Optional IPv6 discovery binary: if set, it is expected to emit
  --ip_discovery_command=VALUE                                     Optional IP discovery binary: if set, it is expected to emit
[vagrant@centos8 bin]$
```
You can see that only the agent has the `ip6_discovery_command` flag.

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot bump-ee

**Comment by GoelDeepak** [comment]
@mesosphere-mergebot label [Ready For Review]

**Comment by asridharan** [comment]
@orsenthil can we merge this. I think it has the pre-req number of ship its.

**Comment by orsenthil** [comment]
@asridharan  - `mergebot/enterprise/build-status/aggregate ` was waiting

**Comment by orsenthil** [comment]
@mesosphere-mergebot label ship-it

**Pull request merged**